### PR TITLE
(PDOC-258) Puppet Strings-ification

### PR DESCRIPTION
**Pull request by eputnam**

### CODE REVIEW

File: REFERENCE.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+Specifies the absolute path to the base location where you want to install IBM Installation Manager. Defaults to `/opt/IBM/InstallationManager`.
+
+Default value: `undef`
+
+##### `manage_user`
+
+Data type: `Any`
+
+Whether or not to manage the user that will be installing the IBM IM. Default: `false`.
```
Comment (pmcmaw) [inline REFERENCE.md]: Default value for this seems to be duplicated.

File: REFERENCE.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+
+Specifies the home directory for the specified user. Required if you're installing in a mode other than 'administrator'.
+
+Default value: `undef`
+
+##### `manage_group`
+
+Data type: `Any`
+
+When in 'group' mode, whether or not to manage the group that will be installing the IBM IM. Default: `false`.
```
Comment (pmcmaw) [inline REFERENCE.md]: Default value is duplicated.

File: REFERENCE.md [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+
+Specifies the timeout for the installation, in seconds. Defaults to 900. Installation Manager can take a long time to install, so if you have issues with Puppet timing out before the installation is complete, you might need to increase this parameter.
+
+Default value: '900'
+
+##### `installation_mode`
+
+Data type: `Any`
+
+Specifies which 'installation mode' you want to use to install the IBM Installation Manager. Values: 'administrator', 'nonadministrator', 'group'. Default: 'administrator'
```
Comment (pmcmaw) [inline REFERENCE.md]: Default value is duplicated.

File: REFERENCE.md [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+Default value: present
+
+#### Parameters
+
+The following parameters are available in the `ibm_pkg` type.
+
+##### `name`
+
+namevar
+
```
Comment (pmcmaw) [inline REFERENCE.md]: Extra new lines.

File: REFERENCE.md [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+If a JDK must be installed separately (as in the case of Websphere Application Server 9), specify the version here. Like 'version', this parameter refers to the number after the underscore in the IBM package name.
+
+##### `imcl_path`
+
+The full path to the imcl executable.
+This is optional. The provider will attempt to locate imcl by parsing /<appDataLocation>/InstallationManager/installed.xml.  If, for some reason, it cannot be discovered or if you need to provide a specific path, you may do so with this parameter.
+
+##### `target`
+
+The full path to install the specified package to. Corresponds to the 'imcl' option '-installationDirectory'
```
Comment (pmcmaw) [inline REFERENCE.md]: Missing . at the end.

File: REFERENCE.md [hunk 6]
```diff
@@ -0,0 +1,233 @@
...
+
+##### `package`
+
+The IBM package name. Example: com.ibm.websphere.IBMJAVA.v71
+This is the first part of the traditional IBM full package name, before the first underscore.
+
+##### `repository`
+
+The full path to the 'repository.config' file for installing this package
```
Comment (pmcmaw) [inline REFERENCE.md]: Missing . at the end.

File: REFERENCE.md [hunk 7]
```diff
@@ -0,0 +1,233 @@
...
+The version of the package. Example: 7.1.2000.20141116_0823
+This is the second part of the traditional IBM full package name, after the first underscore.
+
+##### `options`
+
+Any custom options to pass to the 'imcl' tool for installing the package
+
+##### `user`
+
+The user to run the 'imcl' command as. Defaults to 'root'
```
Comment (pmcmaw) [inline REFERENCE.md]: Missing . at the end and duplicate default value.

File: REFERENCE.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+
+The following parameters are available in the `ibm_installation_manager` class.
+
+##### `deploy_source`
+
+Data type: `Any`
+
+Specifies whether this module should be responsible for deploying the source package for Installation Manager. Valid values are `true` and `false`. Defaults to `false`
+
+Default value: `false`
```
Comment (pmcmaw) [inline REFERENCE.md]: Duplicate default value.

File: REFERENCE.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+
+Default value: `undef`
+
+##### `source_dir`
+
+Data type: `Any`
+
+Specifies the absolute path to the directory to deploy the installer from. (This is the directory containing the `installc` binary.) If you extracted the archive yourself, point this parameter to the extracted archive. Defaults to `/opt/IBM/tmp`.
+
+Default value: `undef`
```
Comment (pmcmaw) [inline REFERENCE.md]: Duplicate default values and they don't match.

File: REFERENCE.md [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+
+##### `target`
+
+Data type: `Any`
+
+Specifies the absolute path to the base location where you want to install IBM Installation Manager. Defaults to `/opt/IBM/InstallationManager`.
+
+Default value: `undef`
```
Comment (pmcmaw) [inline REFERENCE.md]: Duplicate default values and they don't match.

File: REFERENCE.md [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+Specifies options to pass to the installer.
+Default: - administrator mode: `-acceptLicense -s -log/tmp/IM_install.${timestamp}.log.xml -installationDirectory ${target}`. - nonadministrator/group mode: `-acceptLicense -accessRights nonAdmin -s -log /tmp/IM_install.${timestamp}.log.xml`
+
+Default value: `undef`
+
+##### `timeout`
+
+Data type: `Any`
+
+Specifies the timeout for the installation, in seconds. Defaults to 900. Installation Manager can take a long time to install, so if you have issues with Puppet timing out before the installation is complete, you might need to increase this parameter.
```
Comment (pmcmaw) [inline REFERENCE.md]: Duplication of default.

File: REFERENCE.md [hunk 12]
```diff
@@ -0,0 +1,233 @@
...
+Full path to an optional response file to use. The user is responsible for ensuring this file is present.
+
+##### `version`
+
+The version of the package. Example: 7.1.2000.20141116_0823
+This is the second part of the traditional IBM full package name, after the first underscore.
+
+##### `options`
+
+Any custom options to pass to the 'imcl' tool for installing the package
```
Comment (pmcmaw) [inline REFERENCE.md]: Missing `.`

File: REFERENCE.md (hunk 9 above)
Comment (eputnam) [reply REFERENCE.md]: It's also wrong

File: REFERENCE.md (hunk 4 above)
Comment (eputnam) [reply REFERENCE.md]: this is because "name" didn't have a `desc`. It does now.

**Comment by eputnam** [comment]
@pmcmaw , please review my changes.

File: REFERENCE.md [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+
+When in 'group' mode, whether or not to manage the group that will be installing the IBM Installation Manager.
+
+Default value: `false`
+
+##### `group`
+
+Data type: `Any`
+
+Specifies the group to run the installation as. Defaults to `root`.
```
Comment (pmcmaw) [inline REFERENCE.md]: Duplicate default.

**Comment by pmcmaw** [comment]
@eputnam just one remaining duplicate default value, everything else looks great! 👍

**Pull request merged**